posts/2021-03-15-typing-your-actors.adoc

```diff
@@ -23,8 +23,8 @@ Initially, I deliberately chose to use a classic actor implementation so that I
 Obviously, the observable behaviour should stay the same after migration, which consists of:
 
 * a stream for each websocket connection with its own wire data https://github.com/lunatech-labs/lunatech-pointingpoker/blob/master/src/main/scala/com/lunatech/pointingpoker/websocket/WSMessage.scala#L10[domain] forwarding all incoming data to the main application controller (Room Manager) as well as providing an interface to forward outgoing data.
-* the Room manager is responsible for translating from the wire domain to the application https://github.com/lunatech-labs/lunatech-pointingpoker/blob/master/src/main/scala/com/lunatech/pointingpoker/Room.scala#L74-L84[domain] and https://github.com/lunatech-labs/lunatech-pointingpoker/blob/master/src/main/scala/com/lunatech/pointingpoker/RoomManager.scala#L56-L69[route] translated messages to appropriate rooms and manage each room lifecycle.
-* rooms are representations of each planning poker session, holding all necessary https://github.com/lunatech-labs/lunatech-pointingpoker/blob/master/src/main/scala/com/lunatech/pointingpoker/Room.scala#L13-L15[information] and making sure that each participant is up to date with it.
+* the Room manager is responsible for translating from the wire domain to the application https://github.com/lunatech-labs/lunatech-pointingpoker/blob/master/src/main/scala/com/lunatech/pointingpoker/actors/Room.scala#L14-L19[domain] and https://github.com/lunatech-labs/lunatech-pointingpoker/blob/master/src/main/scala/com/lunatech/pointingpoker/actors/RoomManager.scala#L120-L136[route] translated messages to appropriate rooms and manage each room lifecycle.
```

Comment:
Totally up to you, but typically I try to avoid linking to direct lines like this unless you have the hash included. For example `https://github.com/lunatech-labs/lunatech-pointingpoker/blob/f449e67d1adc6fcd0ebf6c71d15c909234109f67/src/main/scala/com/lunatech/pointingpoker/actors/RoomManager.scala#L120-L136`. Mainly because the way you have it, if anything changes, the lines no longer match up. With the hash, you get a snapshot of the code at that time.

You should be able to just go to the file, select the lines you want, and then choose copy permalink. I'd recommend doing that for these to ensure that people in the future are seeing what you want them to see for all these links.

Reply:
Good suggestion actually!

Reply:
Fixed